crowd_anki/github/github_importer.py

```diff
@@ -32,24 +28,15 @@ def import_from_github(self):
         repo, ok = QInputDialog.getText(None, 'Enter GitHub repository',
                                         'Path:', text='<name>/<repository>')
         if repo and ok:
-            self.download_and_import(repo)
+            self.download_and_import_git(repo)
 
-    def download_and_import(self, repo):
+    def download_and_import_git(self, github_repo):
+        repo_parts = github_repo.split("/")
+        repo_dir = Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", repo_parts[-1], "git")
```

Comment:
For the directory I think the ideal option would be to clone things into the place where snapshots go (it's available in config).  The format there is `snapshot_dir/<anki profile name>/<deck name>`

Deck name is a bit tricky - if you follow the convention of naming the repo with the CrowdAnki deck name like here https://github.com/Stvad/Software_Engineering__git then you can use that, but I suspect you don't (being not a fun of those names & all :p)

So another option can be is storing the snapshot destination on per deck basis. Something like:
* If you're importing deck from git repo -> set snapshot destination to the original import directory
* If it's not set -> set it to the default one 
* Use these names on snapshot

Reply:
I'll admit that I didn't understand very well what you meant by the three bullet points above. Please see the commits I just pushed, and how they relate to what you suggest, thank you.

Reply:
Have you had a chance to take a look? Thanks.

Reply:
Should be able to take a closer look today/tomorrow.   

What the comment above was getting at that if the name of repository does not match the name of the deck - then initial import would work but subsequent snapshots will go to a different directory which is problematic. I proposed a way to solve that by changing how we determine the snapshot directory for deck

Reply:
But the name of the repository does not determine the name of the deck: when importing from github, the deck name is taken from `deck.json` So even if I import [the "covid-19-anki-deck](https://github.com/evolverine/covid-19-anki-deck/), the deck I see in Anki is "COVID-19" (in fact, I see "COVID-19_2", "COVID-19_3", etc, but right now that's beyond the point). Indeed, though, the name of the imported repository directory will be `covid-19-anki-deck`, so I see your point.

Nevertheless, the user can rename the deck to something wildly different, and they can also move the cards in and out of it at will. Being driven by the deck name will lead to a multitude of issues.

The way it would make the most sense for me for all this to happen is if each deck had a property called `remote_url` or something like that, which pointed to (either the original, or to the forked) remote repository. Then we'd use this property to determine the directory names for the repositories.
I'm not familiar enough with the Anki codebase to know whether that's something that could be implemented?